### PR TITLE
Laboratory work #2, Shuvar Alena - 19FPL2

### PR DESCRIPTION
**Pull request by alenashuvar**

### CODE REVIEW

File: lab_2/main.py [hunk 1]
```diff
@@ -12,10 +12,14 @@ def tokenize_by_lines(text: str) -> tuple:
...
-    pass
+    if not isinstance(text, str):
+        return ()
+    sentences = text.strip().split('\n')
+    tokenized_sent = tuple(tuple(tokenize(sentence)) for sentence in sentences if len(sentence))
+    return tokenized_sent
 
 
-def create_zero_matrix(rows: int, columns: int) -> list:
+def create_zero_matrix(rows, columns) -> list:
```
Comment (WhiteJaeger) [inline lab_2/main.py]: Please, leave function definition as it is.

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]

File: lab_2/main.py [hunk 2]
```diff
@@ -1,6 +1,28 @@
 """
 Longest common subsequence problem
 """
+from copy import deepcopy
+from tokenizer import tokenize
+
+
+def is_inputs_incorrect(first_sentence_tokens, second_sentence_tokens, plagiarism_threshold=-100):
```
Comment (WhiteJaeger) [inline lab_2/main.py]: Typo: `is` -> `are`.

File: lab_2/main.py [hunk 3]
```diff
@@ -34,7 +69,23 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
...
+
+    lcs_matrix = create_zero_matrix(len(first_sentence_tokens) + 1, len(second_sentence_tokens) + 1)
+    for first_i, first_elem in enumerate(first_sentence_tokens):
+        for sec_i, second_elem in enumerate(second_sentence_tokens):
+            if first_elem == second_elem:
+                lcs_matrix[first_i][sec_i] = lcs_matrix[first_i - 1][sec_i - 1] + 1
+            else:
+                lcs_matrix[first_i][sec_i] = max(lcs_matrix[first_i][sec_i - 1], lcs_matrix[first_i - 1][sec_i])
+
+    del lcs_matrix[-1]
```
Comment (WhiteJaeger) [inline lab_2/main.py]: Please, provide some explanation.

File: lab_2/main.py [hunk 4]
```diff
@@ -34,7 +69,23 @@ def fill_lcs_matrix(first_sentence_tokens: tuple, second_sentence_tokens: tuple)
...
+    for row in lcs_matrix:
+        del row[-1]
```
Comment (WhiteJaeger) [inline lab_2/main.py]: Please, provide some explanation.

**Review by WhiteJaeger — COMMENTED** [review]
Good work!
Just answer to the comments and you are done here.

File: lab_2/main.py (hunk 3 above)
Comment (alenashuvar) [reply lab_2/main.py]: My tests failed when I did not create a column and a row with zeros, but now everything works, and I removed it.

File: lab_2/main.py (hunk 4 above)
Comment (alenashuvar) [reply lab_2/main.py]: I needed to delete zero-row and -column to pass the tests, but now I don't need them anymore

File: lab_2/main.py [hunk 5]
```diff
@@ -1,6 +1,28 @@
...
+
+    if first_is_not_tuple or second_is_not_tuple:
+        return True
+
+    any_not_words_in_first = any(not isinstance(word, str) for word in first_sentence_tokens)
+    any_not_words_in_second = any(not isinstance(word, str) for word in second_sentence_tokens)
+    if any_not_words_in_first or any_not_words_in_second:
+        return True
+
+    if plagiarism_threshold != -100:
```
Comment (WhiteJaeger) [inline lab_2/main.py:24]: Why exactly `-100`?

**Review by WhiteJaeger — APPROVED** [review]
Good job!
You are done here.

File: lab_2/main.py (hunk 5 above)
Comment (alenashuvar) [reply lab_2/main.py:24]: I don't know :)
I just needed a positional argument for the function and -100 is the first thing i thought about

**Review by WhiteJaeger — CHANGES_REQUESTED** [review]
There are [several tests](https://travis-ci.com/github/fipl-hse/2020-2-level-labs/jobs/403905144) that did not pass.
If you need help fixing them - contact me.

**Review by WhiteJaeger — APPROVED** [review]
Very impressive!
You are done here.

**Pull request closed without merging**